data/schema/game_config.cfg

```
@@ -44,6 +44,10 @@
         name="value_type"
         value="empty|value|add|sub|multiply|divide"
     [/type]
+    [type]
+        name="affect_allies_filter"
+        value="empty|yes|no"
```

This allows a literal "empty" value.
```suggestion
        value="|true|false|yes|no"
```

---

Ugh, you actually added support for a literal empty value... Why did you do that?

---

Because empty is an value specific to affect_allies, by default when affect adjacent, abilite affect only same side

---

Sure but that is the empty string not the literal string `empty`...

---

what suggest instead?

---

I made a literal suggestion in the first comment. You can click on commit if you like.

---

i mademy own changes

---

I think this should be more like:

```
[type]
    name="affect_allies_filter"
    [list]
        [element]
            value="true|false|yes|no|empty" # replace empty with whatever we end up settling on
        [/element]
    [/list]
[/type]
```

And then in the code, handle it as a comma-separated list. So you could ask for:

* `affect_allies=yes` – abilities that affect allies and own units
* `affect_allies=empty,no` – any abilities that don't affect allies (but may affect own units)
* `affect_allies=yes,empty` – any ability that affects own units (and possibly allies too)
* etc

Though since there are only a few combos, I suppose there's also the possibility of giving each combo a unique name… but that might just be confusing.

---

The list part is correct. I have no idea what the issue is with using an empty value to represent the exact same in the ability.

---

it's done, but i continue to use 'empty' value